Source/DafnyCore/Plugins/IExecutableBackend.cs

```diff
@@ -169,4 +171,10 @@ public abstract class IExecutableBackend {
   /// Instruments the underlying SinglePassCompiler, if it exists.
   /// </summary>
   public abstract void InstrumentCompiler(CompilerInstrumenter instrumenter, Program dafnyProgram);
+
+  public virtual string JoinNamespaceParts(IEnumerable<string> parts) => string.Join(".", parts);
```

Comment:
One of many bits that goes away if each backend just takes a single-valued option instead of a sequence of parts.

Reply:
Well... actually something like this could reduce a lot of duplication in the back-ends, since they have a lot of `s1 + "." + s2` copied all over the place. But I've removed it now.

Reply:
Oh I don't doubt the utility is useful in many other places, just the wrong place for it. :)